server/handlers/game.go

```diff
@@ -107,7 +108,35 @@ func Commands(game *stycoon.Game, w http.ResponseWriter, req *http.Request) {
 }
 
 func Reports(game *stycoon.Game, w http.ResponseWriter, req *http.Request) {
-	// TODO: all the ****
-	reports, _ := json.Marshal(game.Reports)
-	w.Write(reports)
+	if req.Method != http.MethodGet {
+		log.Warn().Str("method", req.Method).Msg("Unsupported method")
+		http.Error(w, "only GET method is supported", http.StatusBadRequest)
+		return
+	}
+	game.Ready.RLock()
+	defer game.Ready.RUnlock()
+	if stycoon.SeasonChanged(game, req, game.SessionManager) {
+		http.Error(w, "season changed", http.StatusForbidden)
+		return
+	}
+	// TODO: maybe only for logged users?
+	// var err error
+	// _, err = stycoon.LoggedUserFromSession(req, game.SessionManager)
+	// if err != nil {
+	// 	log.Warn().Err(err).Msg("User is not logged in")
+	// 	http.Error(w, "only for logged users", http.StatusForbidden)
+	// 	return
+	// }
+	// TODO: maybe only return player reports?
```

Comment:
Visualization was getting 403 when I required login for reports. So I just commented it out. Idk if we want all reports to be public or if we want to filter them by player.

Reply:
you are right it should be public